integrations/access/accessrequest/app.go

```diff
@@ -136,12 +158,28 @@ func (a *App) run(ctx context.Context) error {
 	return nil
 }
 
+func (a *App) amrAppliesToThisPlugin(amr *accessmonitoringrulesv1.AccessMonitoringRule) bool {
+	if amr.Spec.Notification.Name != a.pluginName {
+		return false
+	}
+	return slices.ContainsFunc(amr.Spec.Subjects, func(subject string) bool {
+		return subject == types.KindAccessRequest
+	})
+}
+
 // onWatcherEvent is called for every cluster Event. It will filter out non-access-request events and
 // call onPendingRequest, onResolvedRequest and on DeletedRequest depending on the event.
 func (a *App) onWatcherEvent(ctx context.Context, event types.Event) error {
-	if kind := event.Resource.GetKind(); kind != types.KindAccessRequest {
-		return trace.Errorf("unexpected kind %s", kind)
+	switch event.Resource.GetKind() {
+	case types.KindAccessMonitoringRule:
+		return trace.Wrap(a.handleAccessMonitoringRule(ctx, event))
+	case types.KindAccessRequest:
+		return trace.Wrap(a.handleAcessRequest(ctx, event))
 	}
+	return trace.Errorf("unexpected kind %s", event.Resource.GetKind())
```

Comment:
```suggestion
	return trace.BadParameter("unexpected kind %s", event.Resource.GetKind())
```

Reply:
@EdwardDowling I have made a few comments here that have been marked resolved but were not resolved. Why not make this change?

You have exactly the same error on line 228 but it correctly uses `BadParameter`

Reply:
That was a mistake, i did not realize this was in multiple places. It should be fixed now.